src/effects/AutoDuck.cpp

```diff
@@ -189,6 +185,30 @@ bool EffectAutoDuck::Process(EffectInstance &, EffectSettings &)
    if (end <= start)
       return false;
 
+   TrackListHolder tempTracks;
+   auto pControlTrack = mControlTrack;
+   // If there is any stretch in the control track, substitute a temporary
+   // rendering before trying to use GetFloats
+   {
+      auto &clips = pControlTrack->GetClips();
+      const auto t0 = pControlTrack->LongSamplesToTime(start);
+      const auto t1 = pControlTrack->LongSamplesToTime(end);
+      const bool hasStretch = any_of(clips.begin(), clips.end(),
```

Comment:
`WaveTrack::Duplicate` does this check for you. We should consider making it a public helper method of WaveTrack to places like these simpler.

Reply:
Pardon me I thought it did: not really, or intricately.
That said, `Duplicate` won't apply stretch ratio if no clip is stretched in the interval. So behaviour would be preserved removing those checks - although it would have a negative impact on performance.